### ISSUE TITLE
latest staging failes pba with private.h error

### ISSUE THREAD

**Issue by SteveEbey73742** [acomminos/wine-pba#45]
When I try to apply the pba patches, modified, or rebased, from the different repos listed in the issues here, fail to apply with an error in private.h.  Anyone else with this issue also?

**Comment by SveSop** [comment]
@SteveEbey73742 [https://github.com/acomminos/wine-pba/issues/46](https://github.com/acomminos/wine-pba/issues/46)

**Comment by SteveEbey73742** [comment]
@SveSop  that is the one I pulled from, trying again, as I see it has an update.

**Comment by SveSop** [comment]
@SteveEbey73742 I updated it like a minute before posting that #46 so most likely you pulled just before i updated :)

If you download my patched source - that is the current wine-dev-git with all staging-git applied + PBA applied, so that is ready to compile.

If you prefer to do it all yourself, you can download wine-dev-git from wineHQ, and staging-git from staging repo, and then the pba-rebased and patch that yourself :)

**Comment by SteveEbey73742** [comment]
Just finished the pull and it patched cleanly. i will continue to check your repo, thanks for your efforts.

**Comment by SteveEbey73742** [comment]
Works, and the fps seems even higher, maybe devel is also doing some cleanup work.  my time test fps benchs are within 5% of what the windows 10 is giving me.

**Comment by SteveEbey73742** [comment]
@SveSop , your repos are throwing 404 errors, are they deleted?

**Comment by Firerat** [comment]
looks like it yeah, only 2 forks now ( see top right , click the number )

**Comment by Firerat** [comment]
just use @acomminos 's master
`git am --3way patches/*patch`
now the fun bit
I will recommend kdiff3-qt , cos it is easy
( if you use kde, then drop the -qt ) # assumes debian, no idea about other distros 

`git mergetool`

dlls/wined3d/wined3d_private.h
needs to end up like this
```
    BYTE bCursorVisible : 1;
    BYTE d3d_initialized : 1;
    BYTE inScene : 1;                   /* A flag to check for proper BeginScene / EndScene call pairs */
    BYTE softwareVertexProcessing : 1;  /* process vertex shaders using software or hardware */
    BYTE filter_messages : 1;
    BYTE use_pba : 1;                   /* A flag to use the persistent buffer allocator for dynamic buffers. */
    BYTE padding : 2;

```
Note, if using kdiff you will have to manually edit the line `BYTE padding : `
brief history

padding was 2, `use_pba` added so padding became 1
upstream removed `BYTE vertexBlendUsed : 1;`so padding became 3
padding = 3, add `use_pba` so padding should become 2
I hope that makes sense

happy fishing, eat well

**Comment by SveSop** [comment]
@SteveEbey73742 Yeah, getting to complicated to follow @Firerat 's directions for me, so decided id just keep it to myself :) 

Just follow his directions, and you will probably get everything fixed in notime :)

**Comment by SteveEbey73742** [comment]
@SveSop your patches were what worked for me. blind in my right eye, makes it hard to do detailed work on computer, without making font really big.  Was hoping to just be able to use your patches, as they are updated with wine staging.

**Comment by Firerat** [comment]
yeah, tbh  I don't know what the incorrect `BYTE padding` would have, probably random
anyway, freshly generated
[PBA_for_wine_5946973_staging_4954f5c.tar.gz](https://github.com/acomminos/wine-pba/files/1848739/PBA_for_wine_5946973_staging_4954f5c.tar.gz)

**Comment by SveSop** [comment]
@SteveEbey73742 No point in having too many repo's doing the same thing really, and since @Firerat is way better than me doing the rebasing and stuff, better have that :)
